### PR TITLE
Fighter cleanup pass 1 by Altimor

### PR DESCRIPTION
**Pull request by ribbanya**
Includes equivalent commits from #636 up to 3cdac6c16058487654f4e9359b926bc1a9b1bbd5.

### CODE REVIEW

**Review by ribbanya — COMMENTED** [review]
[Hiding whitespace](https://github.com/doldecomp/melee/pull/638/files?diff=split&w=1) really helps when reviewing this.

File: Makefile [hunk 1]
```diff
@@ -126,6 +126,7 @@ endif
 # clang-format all source files
 format:
 	$(QUIET) find src include -type f \( -name '*.c' -o -name '*.h' \) -exec $(FORMAT) {} +
+	$(QUIET) tools/newlines.sh
```
Comment (ribbanya) [inline Makefile:129]: Does this work as expected? I don't actually know how make works at all.

Comment (r-burns) [reply Makefile:129]: You need to sh tools/newlines.sh, or add a shebang to it

Comment (ribbanya) [reply Makefile:129]: Okay, I think it needs to be `#!/bin/bash` so I'll add that.

Comment (r-burns) [reply Makefile:129]: Sounds good, but please use `#!/usr/bin/env bash` for better distro compatibility

**Review by r-burns — APPROVED** [review]
Glad I took a closer look at this with whitespace changes hidden, there is some great stuff here. Lots of type refinements in sysdolphin and better names for lots of melee functions. Nice work

**Pull request merged**